app/assets/javascripts/institutions/map_locations.js

```diff
@@ -0,0 +1,170 @@
+
+function handleNoGeolocation(errorFlag) {
+  if (errorFlag == true) {
+    alert("Geolocation service failed.");
+    initialLocation = newyork;
+  } else {
+    alert("Your browser doesn't support geolocation. We've placed you in Siberia.");
+    initialLocation = siberia;
+  }
+  map.setCenter(initialLocation);
+}
+
+
+
+var infowindow = new google.maps.InfoWindow({
+  maxWidth: 160
+});
+
+var markers = new Array();
+
+var iconCounter = 0;
+
+function setMarkers(){
+  // Add the markers and infowindows to the map
+  for (var i = 0; i < locations.length; i++) {  
+    destination = new google.maps.LatLng(locations[i][1], locations[i][2]);
+    var marker = new google.maps.Marker({
+      position: destination,
+      map: map,
+      icon: icons[iconCounter]
+    });
+
+    markers.push(marker);
+
+    google.maps.event.addListener(marker, 'click', (function(marker, i) {
+      return function() {
+        infowindow.setContent(locations[i][0]);
+        infowindow.open(map, marker);
+      }
+    })(marker, i));
+    
+    iconCounter++;
+    // We only have a limited number of possible icon colors, so we may have to restart the counter
+    if(iconCounter >= iconsLength) {
+    	iconCounter = 0;
+    }
+
+    //calculate distances
+    //console.log('initialLocation = ' +initialLocation);
+    calculateDistances(initialLocation, destination);
+    //console.log(results);
+  }
+}
+
+function autoCenter() {
+  //  Create a new viewpoint bound
+  var bounds = new google.maps.LatLngBounds();
+  //  Go through each...
+  for (var i = 0; i < markers.length; i++) {  
+			bounds.extend(markers[i].position);
+  }
+  //  Fit these bounds to the map
+  map.fitBounds(bounds);
+}
+autoCenter();
+
+function calculateDistances(origin, destination) {
+  var service = new google.maps.DistanceMatrixService();
+  service.getDistanceMatrix(
+    {
+      origins: [origin],
+      destinations: [destination],
+      travelMode: google.maps.TravelMode.WALKING,
+      unitSystem: google.maps.UnitSystem.IMPERIAL,
+      avoidHighways: false,
+      avoidTolls: false
+    }, callback);
+}
+
+function callback(response, status) {
+  if (status != google.maps.DistanceMatrixStatus.OK) {
+    alert('Error was: ' + status);
+  } else {
+    var origins = response.originAddresses;
+    var destinations = response.destinationAddresses;
+    //var outputDiv = document.getElementById('outputDiv');
+    //outputDiv.innerHTML = '';
+    //deleteOverlays();
+    console.log('calling back')
+    console.log(response.rows[0].elements[0].distance.text);
+    /*
+    for (var i = 0; i < origins.length; i++) {
+      var results = response.rows[i].elements;
+      addMarker(origins[i], false);
+      for (var j = 0; j < results.length; j++) {
+        addMarker(destinations[j], true);
+        outputDiv.innerHTML += origins[i] + ' to ' + destinations[j]
+            + ': ' + results[j].distance.text + ' in '
+            + results[j].duration.text + '<br>';
+      }
+    }
+  */
+
```

Comment:
Try to delete any commented out code, if the code is needed we can always go back and look at previous commits.